source/create.ts

```diff
@@ -197,6 +198,7 @@ const create = (defaults: InstanceDefaults): Got => {
 
 	// Pagination
 	const paginateEach = (async function * <T, R>(url: string | URL, options?: OptionsWithPagination<T, R>) {
+		// @ts-expect-error
```

Comment:
This too?

Reply:
I have no idea how to fix it. I just punted it as it's more important to get TS upgraded:

```
source/create.ts:201:51 - error TS2345: Argument of type 'Merge<Options, PaginationOptions<T, R>> | undefined' is not assignable to parameter of type 'Options | undefined'.
  Type 'Merge<Options, PaginationOptions<T, R>>' is not assignable to type 'Options | undefined'.
    Type 'Merge<Options, PaginationOptions<T, R>>' is not assignable to type 'Options'.
      Types of property 'pagination' are incompatible.
        Type '{ transform?: ((response: Response<R>) => T[] | Promise<T[]>) | undefined; filter?: ((item: T, allItems: T[], currentItems: T[]) => boolean) | undefined; ... 4 more ...; stackAllItems?: boolean | undefined; } | undefined' is not assignable to type '{ transform?: ((response: Response<unknown>) => unknown[] | Promise<unknown[]>) | undefined; filter?: ((item: unknown, allItems: unknown[], currentItems: unknown[]) => boolean) | undefined; paginate?: ((response: Response<unknown>, allItems: unknown[], currentItems: unknown[]) => false | Options) | undefined; should...'.
          Type '{ transform?: ((response: Response<R>) => T[] | Promise<T[]>) | undefined; filter?: ((item: T, allItems: T[], currentItems: T[]) => boolean) | undefined; ... 4 more ...; stackAllItems?: boolean | undefined; }' is not assignable to type '{ transform?: ((response: Response<unknown>) => unknown[] | Promise<unknown[]>) | undefined; filter?: ((item: unknown, allItems: unknown[], currentItems: unknown[]) => boolean) | undefined; paginate?: ((response: Response<unknown>, allItems: unknown[], currentItems: unknown[]) => false | Options) | undefined; should...'.
            Types of property 'transform' are incompatible.
              Type '((response: Response<R>) => T[] | Promise<T[]>) | undefined' is not assignable to type '((response: Response<unknown>) => unknown[] | Promise<unknown[]>) | undefined'.
                Type '(response: Response<R>) => T[] | Promise<T[]>' is not assignable to type '(response: Response<unknown>) => unknown[] | Promise<unknown[]>'.
                  Types of parameters 'response' and 'response' are incompatible.
                    Type 'Response<unknown>' is not assignable to type 'Response<R>'.
                      Type 'unknown' is not assignable to type 'R'.
                        'R' could be instantiated with an arbitrary type which could be unrelated to 'unknown'.

201  	let normalizedOptions = normalizeArguments(url, options, defaults.options);
     	                                                ~~~~~~~


Found 1 error.
```

I'll add a TODO comment.

Reply:
TS 3.9 introduced a lot of issues, and it seems many will be fixed in TS 4, so I suggest ignoring it until then.

Reply:
https://github.com/microsoft/TypeScript/issues/33436

Reply:
Simplest repro:

```ts
interface Options<T = unknown> {
    fn: (value: T) => void
}

const normalizeOptions = (options: Options): void => {};

function someOtherFunction<T>(options: Options<T>) {
    normalizeOptions(options);
}
```

Reply:
Replacing `unknown` with `any` seems to fix the issue.

Reply:
Interestingly the example reproduces the error in all TS 3.x versions.